tests/nested/manual/core20-early-config/task.yaml

```diff
@@ -80,17 +80,31 @@ execute: |
         remote.exec "cat /etc/hostname" | MATCH "foo"
         remote.exec "hostname" | MATCH "foo"
 
+        # save netplan config
+        remote.exec "sudo cp /etc/netplan/00-snapd-config.yaml ."
+
+        # Should not have all-* snippets from the base
+        not remote.exec "sudo snap get system system.network.netplan.network.ethernets.all-en"
+        not remote.exec "sudo snap get system system.network.netplan.network.ethernets.all-eth"
+
         # netplan config defaults are applied
-        remote.exec "sudo cat /etc/netplan/0-snapd-defaults.yaml" | MATCH br54
+        remote.exec "sudo cat /etc/netplan/00-snapd-config.yaml" | MATCH br54
         remote.exec "sudo netplan get bridges.br54.dhcp4" | MATCH true
         remote.exec "sudo snap get system system.network.netplan.network.bridges.br54.dhcp4" | MATCH true
         remote.exec "sudo netplan get ethernets.ens3.dhcp4" | MATCH false
         # and updating netplan works
         remote.exec "sudo snap set system system.network.netplan.network.bridges.br54.dhcp4=false"
         remote.exec "sudo netplan get bridges.br54.dhcp4" | MATCH false
         remote.exec "sudo snap get system system.network.netplan.network.bridges.br54.dhcp4" | MATCH false
+        # unsetting cleans the snapd configuration, but not the one from cloud-init
+        remote.exec sudo snap unset system system.network.netplan.network
```

Comment:
mmh, now snapd config has less precedence than the cloud-init one, isn't that a problem too?

Reply:
This is the same behavior as before, cloud-init if run creates a file named `50-cloud-init.yaml` that snapd does not touch now or in the past. The fact that "snap get ..." shows the settings from cloud-init is in fact a bit confusing, I wonder if we should use only `00-snapd-config.yaml` also when showing the network settings. We could consider it the UC netplan database and assume we don't have the full control of all the netplan configuration.

Reply:
If we want to have full control, a snap set for the network would have to erase all current netplan configuration and replace with the new one. Leaving 50-cloud-init.yaml around would imply that part of the configuration might not be set with the "set", but come from elsewhere.

I also worry here about the scenarios using network-manager, where files named `90-NM-xxx.yaml` are created. These files are controlled by NM's netplan plugin, and I'm not sure how it will interact with snapd if we need the full control of the network configuration.